Build errors related to spatial_editor_plugin.cpp

**Operating system or device, Godot version, GPU Model and driver (if graphics related):**
- Windows 10, compiled using MSVC 2017
- macOS Sierra, compiled using Xcode 8

Both are compiled from Godot Git https://github.com/godotengine/godot/commit/2d4f6ef0bf70cabb431be33a272addd0fa8b8321.

The error logs are attached below:

### Windows

```
Compiling ==> editor\plugins\spatial_editor_plugin.cpp
spatial_editor_plugin.cpp
editor\plugins\spatial_editor_plugin.cpp(95): error C2666: 'Math::lerp': 2 overloads have similar conversions
c:\projects\godot\core\math\math_funcs.h(157): note: could be 'float Math::lerp(float,float,float)'
c:\projects\godot\core\math\math_funcs.h(156): note: or       'double Math::lerp(double,double,double)'
editor\plugins\spatial_editor_plugin.cpp(95): note: while trying to match the argument list '(float, float, double)'
editor\plugins\spatial_editor_plugin.cpp(96): error C2666: 'Math::lerp': 2 overloads have similar conversions
c:\projects\godot\core\math\math_funcs.h(157): note: could be 'float Math::lerp(float,float,float)'
c:\projects\godot\core\math\math_funcs.h(156): note: or       'double Math::lerp(double,double,double)'
editor\plugins\spatial_editor_plugin.cpp(96): note: while trying to match the argument list '(float, float, double)'
editor\plugins\spatial_editor_plugin.cpp(99): error C2666: 'Math::lerp': 2 overloads have similar conversions
c:\projects\godot\core\math\math_funcs.h(157): note: could be 'float Math::lerp(float,float,float)'
c:\projects\godot\core\math\math_funcs.h(156): note: or       'double Math::lerp(double,double,double)'
editor\plugins\spatial_editor_plugin.cpp(99): note: while trying to match the argument list '(float, float, double)'
scons: *** [editor\plugins\spatial_editor_plugin.windows.opt.tools.64.obj] Error 2
scons: building terminated because of errors.
```

### macOS

```
Compiling ==> editor/plugins/spatial_editor_plugin.cpp
editor/plugins/spatial_editor_plugin.cpp:95:24: error: call to 'lerp' is ambiguous
        camera_cursor.x_rot = Math::lerp(old_camera_cursor.x_rot, cursor.x_rot, MIN(1.0, p_interp_delta * (1 / orbit_inertia)));
                              ^~~~~~~~~~
core/math/math_funcs.h:156:32: note: candidate function
        static _ALWAYS_INLINE_ double lerp(double p_from, double p_to, double p_weight) { return p_from + (p_to - p_from) * p_weight; }
                                      ^
core/math/math_funcs.h:157:31: note: candidate function
        static _ALWAYS_INLINE_ float lerp(float p_from, float p_to, float p_weight) { return p_from + (p_to - p_from) * p_weight; }
                                     ^
editor/plugins/spatial_editor_plugin.cpp:96:24: error: call to 'lerp' is ambiguous
        camera_cursor.y_rot = Math::lerp(old_camera_cursor.y_rot, cursor.y_rot, MIN(1.0, p_interp_delta * (1 / orbit_inertia)));
                              ^~~~~~~~~~
core/math/math_funcs.h:156:32: note: candidate function
        static _ALWAYS_INLINE_ double lerp(double p_from, double p_to, double p_weight) { return p_from + (p_to - p_from) * p_weight; }
                                      ^
core/math/math_funcs.h:157:31: note: candidate function
        static _ALWAYS_INLINE_ float lerp(float p_from, float p_to, float p_weight) { return p_from + (p_to - p_from) * p_weight; }
                                     ^
editor/plugins/spatial_editor_plugin.cpp:99:27: error: call to 'lerp' is ambiguous
        camera_cursor.distance = Math::lerp(old_camera_cursor.distance, cursor.distance, MIN(1.0, p_interp_delta * (1 / translation_inertia)));
                                 ^~~~~~~~~~
core/math/math_funcs.h:156:32: note: candidate function
        static _ALWAYS_INLINE_ double lerp(double p_from, double p_to, double p_weight) { return p_from + (p_to - p_from) * p_weight; }
                                      ^
core/math/math_funcs.h:157:31: note: candidate function
        static _ALWAYS_INLINE_ float lerp(float p_from, float p_to, float p_weight) { return p_from + (p_to - p_from) * p_weight; }
                                     ^
3 errors generated.
scons: *** [editor/plugins/spatial_editor_plugin.osx.opt.tools.64.o] Error 1
scons: building terminated because of errors.
```

The build doesn't seem to have any errors on Ubuntu 14.04 (using GCC).

--- a/editor/plugins/spatial_editor_plugin.cpp
+++ b/editor/plugins/spatial_editor_plugin.cpp
@@ -92,11 +92,11 @@ void SpatialEditorViewport::_update_camera(float p_interp_delta) {
 	Cursor old_camera_cursor = camera_cursor;
 	camera_cursor = cursor;
 
-	camera_cursor.x_rot = Math::lerp(old_camera_cursor.x_rot, cursor.x_rot, MIN(1.0, p_interp_delta * (1 / orbit_inertia)));
-	camera_cursor.y_rot = Math::lerp(old_camera_cursor.y_rot, cursor.y_rot, MIN(1.0, p_interp_delta * (1 / orbit_inertia)));
+	camera_cursor.x_rot = Math::lerp(old_camera_cursor.x_rot, cursor.x_rot, MIN(1.f, p_interp_delta * (1 / orbit_inertia)));
+	camera_cursor.y_rot = Math::lerp(old_camera_cursor.y_rot, cursor.y_rot, MIN(1.f, p_interp_delta * (1 / orbit_inertia)));
 
-	camera_cursor.pos = old_camera_cursor.pos.linear_interpolate(cursor.pos, MIN(1.0, p_interp_delta * (1 / translation_inertia)));
-	camera_cursor.distance = Math::lerp(old_camera_cursor.distance, cursor.distance, MIN(1.0, p_interp_delta * (1 / translation_inertia)));
+	camera_cursor.pos = old_camera_cursor.pos.linear_interpolate(cursor.pos, MIN(1.f, p_interp_delta * (1 / translation_inertia)));
+	camera_cursor.distance = Math::lerp(old_camera_cursor.distance, cursor.distance, MIN(1.f, p_interp_delta * (1 / translation_inertia)));
 
 	if (p_interp_delta == 0 || is_freelook_active()) {
 		camera_cursor = cursor;

